### ISSUE TITLE
Yarn link and install doesn't work

### ISSUE THREAD

**Issue by StarpTech** [yarnpkg/yarn#3257]
There is definitely an issue with this. Look at [issue ](https://github.com/yarnpkg/yarn/issues/1297) it's 7 Months old and without a response from the yarn team.

**Comment by voxsim** [comment]
I will look into this. Let me know if you have additional informations; thanks ;)

**Comment by StarpTech** [comment]
Easy to reproduce

1. Link package
2. Install package via link
3. Run `yarn`

Output:
```
yarn
yarn install v0.23.2
[1/4] Resolving packages...
error Received malformed response from registry for "local-test". The registry may be down.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```


```

**Comment by athaax** [comment]
On a related note, but perhaps a separate issue. I experience the following behavior: 
1. `yarn link "my-local-module"`
2. `yarn unlink 'my-local-module'
3. `yarn install`
4. `success Already up-to-date`

In other words, yarn doesn't seem to be picking up the fact that when I unlink a local module, it's out of sync. I have to run `yarn install --force` to get it back, or `yarn add my-local-module@1.2.3` to restore it. 

Shouldn't running `yarn install` be sufficient to restore that package?

**Comment by rally25rs** [comment]
@bestander triage; could be closed as duplicate of #1297 and continue tracking the issue there.

**Comment by StarpTech** [comment]
@rally25rs I hope the issue does not goes down because the referenced issue is 7 months old.
